x/lend/keeper/gov.go

```diff
@@ -51,13 +51,86 @@ func (k *Keeper) AddWhitelistedAssetRecords(ctx sdk.Context, records ...types.As
 }
 
 func (k *Keeper) UpdateWhitelistedAssetRecords(ctx sdk.Context, msg types.Asset) error {
+	asset, found := k.GetAsset(ctx, msg.Id)
+	if !found {
+		return types.ErrorAssetDoesNotExist
+	}
+	if msg.Name != "" {
+		asset.Name = msg.Name
+	}
+	if msg.Denom != "" {
```

Comment:
Same why not check for len(msg.Denom)>0. check at all places

Reply:
created a new PR , resolving the merge conflicts and considering above suggestions.
https://github.com/comdex-official/comdex/pull/142
hence, closing this PR